cmd/kubeadm/app/phases/kubelet/flags_test.go

```diff
@@ -189,3 +190,81 @@ func TestGetNodeNameAndHostname(t *testing.T) {
 		})
 	}
 }
+
+func TestReadKubeadmFlags(t *testing.T) {
+	tests := []struct {
+		name          string
+		fileContent   string
+		expectedValue []string
+		expectError   bool
+	}{
+		{
+			name:          "valid kubeadm flags with container-runtime-endpoint",
+			fileContent:   `KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock --pod-infra-container-image=registry.k8s.io/pause:3.10"`,
+			expectedValue: []string{"--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock", "--pod-infra-container-image=registry.k8s.io/pause:3.10"},
+			expectError:   false,
+		},
+		{
+			name:          "no container-runtime-endpoint found",
+			fileContent:   `KUBELET_KUBEADM_ARGS="--pod-infra-container-image=registry.k8s.io/pause:3.10"`,
+			expectedValue: []string{"--pod-infra-container-image=registry.k8s.io/pause:3.10"},
+			expectError:   true,
+		},
+		{
+			name:          "no KUBELET_KUBEADM_ARGS line",
+			fileContent:   `# This is a comment, no args here`,
+			expectedValue: nil,
+			expectError:   true,
+		},
+		{
+			name:          "invalid file format",
+			fileContent:   "",
+			expectedValue: nil,
+			expectError:   true,
+		},
+		{
+			name:          "Multiple flags with mixed equals and no equals",
```

Comment:
```suggestion
			name:          "multiple flags with mixed equals and no equals",
```

test names should start either with small or capital letter everywhere.